### PR TITLE
Re-send ChannelUpdate on reconnection (vol 2)

### PR DESCRIPTION
**Pull request by pm47**
Previous implementation in #1317 wasn't working because of a bug in the
transition. Added a test and fixed it.

cc @t-bast @btcontract

### CODE REVIEW

File: eclair-core/src/test/scala/fr/acinq/eclair/channel/states/e/OfflineStateSpec.scala [hunk 1]
```diff
@@ -534,4 +534,69 @@ class OfflineStateSpec extends TestkitBaseClass with StateTestsHelperMethods {
...
+    awaitCond(alice.stateName == OFFLINE)
+    awaitCond(bob.stateName == OFFLINE)
+    sender.send(alice, INPUT_RECONNECTED(alice2bob.ref, aliceInit, bobInit))
+    sender.send(bob, INPUT_RECONNECTED(bob2alice.ref, bobInit, aliceInit))
+    alice2bob.expectMsgType[ChannelReestablish]
+    bob2alice.expectMsgType[ChannelReestablish]
+    bob2alice.forward(alice)
+    alice2bob.forward(bob)
+
+    //  point the channel still isn't deeply buried: channel_update isn't sent again
```
Comment (t-bast) [inline eclair-core/src/test/scala/fr/acinq/eclair/channel/states/e/OfflineStateSpec.scala]: nit:
```suggestion
    //  at this point the channel still isn't deeply buried: channel_update isn't sent again
```

File: eclair-core/src/test/scala/fr/acinq/eclair/channel/states/e/OfflineStateSpec.scala [hunk 2]
```diff
@@ -534,4 +534,69 @@ class OfflineStateSpec extends TestkitBaseClass with StateTestsHelperMethods {
...
+    alice2bob.expectNoMsg()
+    bob2alice.expectNoMsg()
+
+    // funding tx gets 6 confirmations, channel is private so there is no announcement sigs
```
Comment (t-bast) [inline eclair-core/src/test/scala/fr/acinq/eclair/channel/states/e/OfflineStateSpec.scala]: You should have an `alice2bob.expectNoMsg()` to verify that no announcement sig is sent, don't you?

Comment (pm47) [reply eclair-core/src/test/scala/fr/acinq/eclair/channel/states/e/OfflineStateSpec.scala]: We receive `ChannelUpdate` instead.

Comment (t-bast) [reply eclair-core/src/test/scala/fr/acinq/eclair/channel/states/e/OfflineStateSpec.scala]: Isn't there a way to expect no message of type X?

**Pull request merged**